### PR TITLE
Stock Market Prediction (Deep Learning Algorithms-neural networks with fine tuning and advanced optimization methods)

### LINKED ISSUE #346: 💡[Feature]: Stock Market Prediction (Deep Learning Algorithms-neural networks with fine tuning and advanced optimization methods)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

we will use high accurate and effiecient algo for stock market predictions and use optimizations algorithms

### Use Case

we can predict the stock and analysis the market condition according to demand and supply

### Benefits

help people to do business.

### Add ScreenShots

No thanks

### Priority

Medium

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by ranamanish674zu**
## Description
Stock Market Prediction (Deep Learning Algorithms-neural networks with fine tuning and advanced optimization methods)
techniques-
Neural net, early stopping and other optimization methods .
Fixes #346 
here are some neccessory library-
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MinMaxScaler
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.callbacks import EarlyStopping, ModelCheckpoint
## Type of change

- [ +] Added a new machine learning frameworks, libraries or software.
- [+ ] Documentation update

## Checklist:

- [ +] My code follows the style guidelines of this project
- [+ ] I have performed a self-review of my own code
- [+ ] I have commented my code, particularly in hard-to-understand areas
- [+ ] I have made corresponding changes to the documentation
- [+ ] My changes generate no new warnings
@sanjay-kv complete #346 please chake

### CODE REVIEW

**Comment by sanjay-kv** [comment]
Why did you raise PR even before assigning?

**Comment by ranamanish674zu** [comment]
> Why did you raise PR even before assigning?

sorry sir , but you have assigned and label it , then i raise the pull request

**Comment by ranamanish674zu** [comment]
> Why did you raise PR even before assigning?



> > Why did you raise PR even before assigning?
> 
> sorry sir , but you have assigned and label it , then i raise the pull request

image

**Comment by sanjay-kv** [comment]
my bad
